nifi-commons/nifi-expression-language/src/main/java/org/apache/nifi/attribute/expression/language/evaluation/functions/MathEvaluator.java

```diff
@@ -0,0 +1,138 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.attribute.expression.language.evaluation.functions;
+
+import org.apache.nifi.attribute.expression.language.evaluation.Evaluator;
+import org.apache.nifi.attribute.expression.language.evaluation.NumberEvaluator;
+import org.apache.nifi.attribute.expression.language.evaluation.NumberQueryResult;
+import org.apache.nifi.attribute.expression.language.evaluation.QueryResult;
+import org.apache.nifi.attribute.expression.language.exception.AttributeExpressionLanguageException;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Map;
+
+public class MathEvaluator extends NumberEvaluator {
+
+    private final Evaluator<Number> subject;
+    private final Evaluator<String> methodName;
+    private final Evaluator<Number> optionalArg;
+
+    public MathEvaluator(final Evaluator<Number> subject, final Evaluator<String> methodName, final Evaluator<Number> optionalArg) {
+        this.subject = subject;
+        this.methodName = methodName;
+        this.optionalArg = optionalArg;
+    }
+
+    @Override
+    public QueryResult<Number> evaluate(final Map<String, String> attributes) {
+        final String methodNamedValue = methodName.evaluate(attributes).getValue();
+        if (methodNamedValue == null) {
+            return new NumberQueryResult(null);
+        }
+
+        final Number subjectValue;
+        if(subject != null) {
+            subjectValue = subject.evaluate(attributes).getValue();
+            if(subjectValue == null){
+                return new NumberQueryResult(null);
+            }
+        } else {
+            subjectValue = null;
+        }
+
+        final Number optionalArgValue;
+        if(optionalArg != null) {
+            optionalArgValue = optionalArg.evaluate(attributes).getValue();
+
+            if(optionalArgValue == null) {
+                return new NumberQueryResult(null);
+            }
+        } else {
+            optionalArgValue = null;
+        }
+
+        try {
+            Number executionValue = null;
+
+            if (subjectValue == null){
+                Method method = Math.class.getMethod(methodNamedValue);
+
+                if(method == null) {
+                    throw new AttributeExpressionLanguageException("Cannot evaluate 'math' function because no subjectless method was found with the name:'" + methodNamedValue + "'");
+                }
+
+                executionValue = (Number) method.invoke(null);
+            } else if(optionalArg == null) {
+                boolean subjectIsDecimal = subjectValue instanceof Double;
+                Method method = Math.class.getMethod(methodNamedValue, subjectIsDecimal ? double.class : long.class);
```

Comment:
If I use:
````
${filename:math('exp')}
````
I'll get the following exception:
````
org.apache.nifi.attribute.expression.language.exception.AttributeExpressionLanguageException: Unable to calculate math function value
	at org.apache.nifi.attribute.expression.language.evaluation.functions.MathEvaluator.evaluate(MathEvaluator.java:129) ~[nifi-expression-language-1.1.0-SNAPSHOT.jar:1.1.0-SNAPSHOT]
	at org.apache.nifi.attribute.expression.language.Query.evaluate(Query.java:568) ~[nifi-expression-language-1.1.0-SNAPSHOT.jar:1.1.0-SNAPSHOT]
	at org.apache.nifi.attribute.expression.language.Query.evaluateExpression(Query.java:386) ~[nifi-expression-language-1.1.0-SNAPSHOT.jar:1.1.0-SNAPSHOT]
	at org.apache.nifi.attribute.expression.language.StandardPreparedQuery.evaluateExpressions(StandardPreparedQuery.java:48) ~[nifi-expression-language-1.1.0-SNAPSHOT.jar:1.1.0-SNAPSHOT]
	at org.apache.nifi.attribute.expression.language.StandardPropertyValue.evaluateAttributeExpressions(StandardPropertyValue.java:152) ~[nifi-expression-language-1.1.0-SNAPSHOT.jar:1.1.0-SNAPSHOT]
	at org.apache.nifi.attribute.expression.language.StandardPropertyValue.evaluateAttributeExpressions(StandardPropertyValue.java:133) ~[nifi-expression-language-1.1.0-SNAPSHOT.jar:1.1.0-SNAPSHOT]
	at org.apache.nifi.processors.attributes.UpdateAttribute.executeActions(UpdateAttribute.java:496) ~[na:na]
	at org.apache.nifi.processors.attributes.UpdateAttribute.onTrigger(UpdateAttribute.java:377) ~[na:na]
	at org.apache.nifi.processor.AbstractProcessor.onTrigger(AbstractProcessor.java:27) ~[nifi-api-1.1.0-SNAPSHOT.jar:1.1.0-SNAPSHOT]
	at org.apache.nifi.controller.StandardProcessorNode.onTrigger(StandardProcessorNode.java:1064) ~[nifi-framework-core-1.1.0-SNAPSHOT.jar:1.1.0-SNAPSHOT]
	at org.apache.nifi.controller.tasks.ContinuallyRunProcessorTask.call(ContinuallyRunProcessorTask.java:136) [nifi-framework-core-1.1.0-SNAPSHOT.jar:1.1.0-SNAPSHOT]
	at org.apache.nifi.controller.tasks.ContinuallyRunProcessorTask.call(ContinuallyRunProcessorTask.java:47) [nifi-framework-core-1.1.0-SNAPSHOT.jar:1.1.0-SNAPSHOT]
	at org.apache.nifi.controller.scheduling.TimerDrivenSchedulingAgent$1.run(TimerDrivenSchedulingAgent.java:132) [nifi-framework-core-1.1.0-SNAPSHOT.jar:1.1.0-SNAPSHOT]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [na:1.8.0_77]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308) [na:1.8.0_77]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180) [na:1.8.0_77]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294) [na:1.8.0_77]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0_77]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [na:1.8.0_77]
	at java.lang.Thread.run(Thread.java:745) [na:1.8.0_77]
Caused by: java.lang.NoSuchMethodException: java.lang.Math.exp(long)
	at java.lang.Class.getMethod(Class.java:1786) ~[na:1.8.0_77]
	at org.apache.nifi.attribute.expression.language.evaluation.functions.MathEvaluator.evaluate(MathEvaluator.java:82) ~[nifi-expression-language-1.1.0-SNAPSHOT.jar:1.1.0-SNAPSHOT]
	... 19 common frames omitted
````
I know this is because I didn't use ``toDecimal()`` but I am wondering if you were not expecting to throw the custom exception just few lines after? IMO just throwing the current exception may not be very clear to users trying to understand why it didn't work. I think users may forget to cast arguments and this will probably be the most common mistake, it could be useful to remember this "requirement" in the exception message. Thoughts?